core/state/statedb.go

```diff
@@ -66,6 +66,7 @@ func (self *StateDB) AddLog(log *Log) {
 	log.TxHash = self.thash
 	log.BlockHash = self.bhash
 	log.TxIndex = uint(self.txIndex)
+	log.Index = uint(len(self.logs))
```

Comment:
Shit. Even that is wrong. I meant to do `len(self.logs[self.thash])` (it's the len of  that because it's set prior to adding the log)